src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,21 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
         c.setTime(date);
+        String timeZone = getCustomIdOfTimeZone(sessionTimeZone);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
```

Comment:
From a quick [search](https://github.com/TEAMMATES/teammates/issues?utf8=✓&q=is%3Aissue%20noon%20) in the issue tracker: #4667